### PR TITLE
Update Server's README with undocumented options for RoPE, YaRN, and KV cache quantization

### PR DESCRIPTION
**Pull request by K-Mistele**
I recently updated my LLama.cpp and found that there are a number of server CLI options which are not described in the README including for RoPE, YaRN, and KV cache quantization as well as flash attention. 

This PR updates the README for the server to include these options as well as their possible values. Future work to be done here (not in this PR, but in a future one by me) will probably include a guide to using RoPE/YaRN scaling since the configuration of these parameters is non-obvious and requires going through some older issues to figure out what expected values might look like.

### CODE REVIEW

File: examples/server/README.md [hunk 1]
```diff
@@ -62,6 +62,18 @@ page cache before using this. See https://github.com/ggerganov/llama.cpp/issues/
...
+- `--rope-scaling` : RoPE scaling method. Defaults to linear unless otherwise specified by the model. Options are `none`, `linear`, `yarn`
+- `--rope-freq-base N` : RoPE frequency base (default: loaded from model)
+- `--rope-freq-scale N`: RoPE frequency scaling factor, expands context by a factor of 1/N (e.g. 0.25)
+- `--yarn-ext-factor N` : YaRN: extrapolation mix factor (Default: 1.0, 0.0 = full interpolation)
+- `--yarn-attn-factor N` : YaRN: scale sqrt(t) or attention magnitude (default: 1.0)
+- `--yarn-beta-slow N`: YaRN: High correction dim or alpha (default: 1.0)
+- `--yarn-beta-fast N`: YaRN: low correction dim or beta (default: 32.0)
+- `--pooling` : Pooling type for embeddings, use model default if unspecified. Options are `none`, `mean`, `cls`
+- `-dt N`, `--defrag-thold N`: KV cache defragmentation threshold (default: -1.0, < 0 = disabled)
+- `-fa`, `--flash-attn` : enable flash attention (default: disabled).
```
Comment (phymbert) [inline examples/server/README.md:74]: Should we consider adding this to the hot topic section, @ggerganov ?

Comment (K-Mistele) [reply examples/server/README.md:74]: probably worth noting that the flash attn support is somewhat experimental and may not work on all platforms. 

Also I have been meaning to put together a guide on RoPE / YaRN / grouped attn (self-extend) configuration for different models & length extension factors but haven't gotten to it yet

**Comment by K-Mistele** [comment]
bump

**Review by Jeximo — CHANGES_REQUESTED** [review]
There seems to be a trailing whitespace blocking editor.config

**Review by ggerganov — APPROVED** [review]

**Pull request merged**